File: praksa-vanjateodorovic-backend/app/models/Post.scala
package models

import dtos.{PostDTO}
import play.api.libs.json.{Format, Json}

import java.time.Instant
import java.util.{Date}

case class Post(
                postId : Option[Long],
                userId : Long,
                content : String,
                timeOfCreation : Date
               )

object Post {
  implicit val format: Format[Post] = Json.format[Post]

  implicit def postDTOToPost(postDTO : PostDTO) : Post = Post(postDTO.postId, postDTO.userId, postDTO.content, new Date())

  //  implicit val timeFormat: Format[Timestamp] = Json.format[Timestamp]

}

File: frontend-angular/src/app/modules/model/post.ts
export class Post{
    postId: number = 0;
    userId: number = 0;
    content: String = "";
    timeOfCreation!: Date;
    
    public constructor(obj?: any) {
        if (obj) {
            this.postId = obj.postId;
            this.userId = obj.userId;
            this.content = obj.content;
            this.timeOfCreation = obj.timeOfCreation;
        }
    }
}